compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -451,19 +451,21 @@ impl<'tcx> WfPredicates<'tcx> {
                 GenericArgKind::Const(ct) => {
                     match ct.kind() {
                         ty::ConstKind::Unevaluated(uv) => {
-                            let obligations = self.nominal_obligations(uv.def.did, uv.substs);
-                            self.out.extend(obligations);
-
-                            let predicate =
-                                ty::Binder::dummy(ty::PredicateKind::ConstEvaluatable(ct));
-                            let cause = self.cause(traits::WellFormed(None));
-                            self.out.push(traits::Obligation::with_depth(
-                                self.tcx(),
-                                cause,
-                                self.recursion_depth,
-                                self.param_env,
-                                predicate,
-                            ));
+                            if !ct.has_escaping_bound_vars() {
```

Comment:
While this fixes the ICE this is probably unsound around inline consts and const generics.

We should not skip logic if there are excess bound vars outside of possibly diagnostics. The right fix here would be to find out where the binder was skipped and stop doing that. This will require wrappig some function arguments in binders at call sites to the current function.

Reply:
Ah... this is "ok" because we can't make it unsound at present and fixing it in the way I described would cause errors about things having to live for 'static.

Please add a comment explaining why not adding these obligations is ok

Reply:
Yeah, I'm not really sure how I feel about this change overall. I don't like adding yet another place that we're not registering WF with bound vars.

Reply:
I'm willing to work on another solution, but I am not sure where to start for that.

Reply:
We could keep the binders around, but then we'd do higher ranked trait solving which just tells us that all lifetimes must be 'static. So basically turning the ICE into an error